notebooks/220-license-plate-recognition/216-license-plate-recognition.ipynb

```diff
@@ -0,0 +1,405 @@
+    "#### Process the Image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "input_img = image_loader(image_path=\"data/example.png\", augmentation=\"None\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Input the Data and Get the Result"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# an auxiliary blob that is needed for correct decoding.\n",
+    "# set this to [1, 1, ..., 1].\n",
+    "auxiliary_blob = np.array([1] * 88)\n",
+    "# resize it to fit the shape: [88, 1].\n",
+    "auxiliary_blob = np.resize(auxiliary_blob, (88, 1))\n",
+    "\n",
+    "# get the result.\n",
+    "result = compiled_model([input_img, auxiliary_blob])[output_layer]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Match the Output Array into Correct Recognition Results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def result_to_string(result: np.ndarray) -> str:\n",
+    "    \"\"\"\n",
+    "    Match the output array into correct recognition results.\n",
+    "\n",
+    "    Args:\n",
+    "        result (np.ndarray): The output of the model. The dimension is 1x88x1x1.\n",
+    "\n",
+    "    Returns:\n",
+    "        str: The license plate recognition results.\n",
+    "    \"\"\"\n",
+    "    # result is an encoded vector of floats, whose shape is [1, 88, 1, 1].\n",
+    "    vector_size = 88\n",
+    "    \n",
+    "    # each float is an integer number encoding a character according to this dictionary\n",
+    "    match_dictionary = open(\"data/match_dictionary.txt\").read().splitlines()\n",
+    "    str_list = list()\n",
+    "    \n",
+    "    # flatten result so it can be just accessed with result[iter]\n",
+    "    result = result.flatten()\n",
+    "    \n",
+    "    for iter in range(vector_size):\n",
+    "        if result[iter] != -1:\n",
+    "            str_list.append(match_dictionary[int(result[iter])])\n",
+    "        else:\n",
+    "            break\n",
+    "            \n",
+    "    ans = \"\".join(str_list)\n",
+    "    return ans"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Print out the result"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get the result.\n",
+    "plate_recognition = result_to_string(result)\n",
+    "\n",
+    "# print out the match result.\n",
+    "print(f\"The license plate recognition result is: {plate_recognition}\")\n",
+    "\n",
+    "# compare to the origin image.\n",
+    "compare = cv2.imread(\"data/example.png\")\n",
```

Comment:
it would be nice to load the original image above and re-use it here instead of loading it once again, consider if you want to change the path to something else - you would need to replace it in 2 places